storage/google/cloud/storage/client.py

```diff
@@ -477,6 +477,37 @@ def list_buckets(
             extra_params=extra_params,
         )
 
+    def get_blob_from_uri(self, uri):
+        """Get a constructor for blob object by URI.
+
+         :type uri: str
+         :param uri: The blob uri pass to get blob object.
+
+         :rtype: :class:`google.cloud.storage.blob.Blob`
+         :returns: The blob object created.
+         """
+        scheme, netloc, path, query, frag = urlsplit(uri)
+        if scheme != "gs":
+            raise ValueError("URI scheme must be gs")
+
+        bucket = Bucket(self, name=netloc)
+        return Blob(path[1:], bucket)
```

Comment:
@frankyn, @tswast  Given that the names for these methods start with `get_`, should we be making the API calls to load the objects?

Reply:
Yes. `get_` implies actually making a GET request. This should call `.reload()`.